### PR TITLE
Added more tests for CLI commands. 

### PR DESCRIPTION
**Pull request by Gitstar-OC**
This PR 

- Adds more tests in the `tests/test_cli_loads.py` folder to test the other commands 
- Updates the contributing file and adds testing part in it instead of ```## Tests HAHAHAHAHAHAHA good one```

### CODE REVIEW

**Comment by Gitstar-OC** [comment]
As asked in #7

**Comment by bboynton97** [comment]
hi @Gitstar-OC! thank you for the contribution!

in the future when opening a pull request, be sure to merge from a branch other than `main` even on your fork. it allows the maintainers to easily add to your contribution and help get a PR past the finish line :)

since i can't push to this branch, i'm including my changes as notes in the review. if you can commit these, i'll ask @tcdent to review again (since my work is in this PR now) and merge

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -36,4 +36,11 @@ Adding tools is easy once you understand the project structure. A few things nee
 4. Manually test your tool integration by running `agentstack tools add <your_tool>` and ensure it behaves as expected.
 
 ## Tests
```
Comment (bboynton97) [inline CONTRIBUTING.md:38]:
```markdown
## Tests
Tests are written in `tests` folder with the cli tests being in `tests/test_cli_loads.py`. 

Testing is sparse at the moment but contributions are very welcome! 🙏

You can run tests by first installing test dependencies with
```bash
pip install ".[test]"
```

Then running tox with
```bash
tox
```
```

Comment (bboynton97) [reply CONTRIBUTING.md:38]: remove the escape characters on the ```s, the formatting got messed up on github :)

File: tests/test_cli_loads.py [hunk 2]
```diff
@@ -43,6 +43,50 @@ def test_init_command(self):
 
         # Clean up
         shutil.rmtree(test_dir)
+    def test_generate_agent_command(self):
```
Comment (bboynton97) [inline tests/test_cli_loads.py:46]:
```python
    # TODO: this is definitely not a clean way to test the CLI and
    # TODO: should be done more elegantly. For now, this allows
    # TODO: the tests to run sequentially and maintain state between
    # TODO: them without creating conflicting projects.
    def test_project_build_command(self):
        """Test the 'init' command to create a project directory."""
        # Ensure the directory doesn't exist from previous runs
        if self.TEST_DIR.exists():
            shutil.rmtree(self.TEST_DIR)

        result = self.run_cli("init", str(self.TEST_DIR), "--no-wizard")
        self.assertEqual(result.returncode, 0)
        self.assertTrue(self.TEST_DIR.exists())

        """Test the 'generate agent' command."""
        agent_name = "test_agent"
        result = self.run_cli("generate", "agent", agent_name, cwd=self.TEST_DIR)
        self.assertEqual(result.returncode, 0)
        # Verify that the agent is added to agents.yaml
        agents_config = self.TEST_DIR / Path("src/config/agents.yaml")
        self.assertTrue(agents_config.exists())
        with open(agents_config, 'r') as f:
            content = f.read()
            self.assertIn(agent_name, content)

        """Test the 'generate task' command."""
        task_name = "test_task"
        result = self.run_cli("generate", "task", task_name, cwd=self.TEST_DIR)
        self.assertEqual(result.returncode, 0)
        # Verify that the task is added to tasks.yaml
        tasks_config = self.TEST_DIR /Path("src/config/tasks.yaml")
        self.assertTrue(tasks_config.exists())
        with open(tasks_config, 'r') as f:
            content = f.read()
            self.assertIn(task_name, content)

        """Test the 'tools list' command."""
        result = self.run_cli("tools", "list", cwd=self.TEST_DIR)
        self.assertEqual(result.returncode, 0)
        self.assertIn("Available AgentStack Tools:", result.stdout)

        """Test the 'tools add' command."""
        tool_name = "ftp"
        result = self.run_cli("tools", "add", tool_name, cwd=self.TEST_DIR)
        self.assertEqual(result.returncode, 0)
        self.assertIn(f"Tool {tool_name} added", result.stdout)
        # Clean up: remove the tool
        self.run_cli("tools", "remove", tool_name)

        """Test the 'run' command."""
        result = self.run_cli("run", cwd=self.TEST_DIR)
        self.assertEqual(result.returncode, 0)

        # Clean up
        shutil.rmtree(self.TEST_DIR)
```

**Review by braelyn-ai — CHANGES_REQUESTED** [review]
^

**Comment by Gitstar-OC** [comment]
Hey sorry for the problems caused I will solve them.

File: CONTRIBUTING.md (hunk 1 above)
Comment (Gitstar-OC) [reply CONTRIBUTING.md:38]: Sure, working on this will solve this :)

**Comment by Gitstar-OC** [comment]
Update worked on the compelete part, will create a pr shortly after a few hours. The work is done

File: tests/test_cli_loads.py (hunk 2 above)
Comment (Gitstar-OC) [reply tests/test_cli_loads.py]: Updated the cli file. :)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -36,4 +36,11 @@ Adding tools is easy once you understand the project structure. A few things nee
...
-HAHAHAHAHAHAHA good one
\ No newline at end of file
+Tests are written in `tests` folder with the cli tests being in `tests/test_cli_loads.py`. 
+
+Currently some of the tests will not work but you can run the tests by 
+
+```
+cd tests
+python -m unittest discover
```
[the remark this answers is not in the document]
Comment (Gitstar-OC) [reply CONTRIBUTING.md]: Updated the contributing.md file and tox is added in it now.

File: CONTRIBUTING.md (hunk 1 above)
Comment (Gitstar-OC) [reply CONTRIBUTING.md:41]: Done

File: tests/test_cli_loads.py [hunk 4]
```diff
@@ -43,6 +43,62 @@ def test_init_command(self):
 
         # Clean up
         shutil.rmtree(test_dir)
+
```
Comment (bboynton97) [inline tests/test_cli_loads.py:46]: please add/merge these into the file too

```python
    def setUp(self):
        self.TEST_DIR = Path("test_project")

    def run_cli(self, *args, cwd=None):
        """Helper method to run the CLI with arguments."""
        result = subprocess.run(
            [*self.CLI_ENTRY, *args],
            capture_output=True,
            text=True,
            cwd=cwd
        )
        return result
```

**Comment by bboynton97** [comment]
one more edit required. be sure to run `tox` to test locally and ensure it works

File: tests/test_cli_loads.py (hunk 4 above)
Comment (Gitstar-OC) [reply tests/test_cli_loads.py:59]: Done.

**Comment by Gitstar-OC** [comment]
All of the things are completed, also ran the tests. Sorry that you faced some problems becuase I did not create a new branch will create a new one for other issues, could I get to work a little bit on frontend if that's possible ?  

image

**Tests were successful**

**Comment by Gitstar-OC** [comment]
I don't understand why are this workflows failing ? They are working if I run the tests in the development mode and failing in production. 

I will check the error and fix them don't worry.

**Comment by potpiedev** [comment]
Starting Review, should be available shortly

This PR addresses and fixes medium-severity vulnerabilities identified in the codebase. The changes include security patches to resolve the identified issues.

**Pull request closed without merging**